conda/notices/core.py

```diff
@@ -66,19 +80,21 @@ def notices(func):
     Args:
         func: Function to be decorated
     """
-
     @wraps(func)
     def wrapper(*args, **kwargs):
-        return_value = func(*args, **kwargs)
-
         if is_channel_notices_enabled(context):
-            display_notices(
+            channel_notice_set = retrieve_notices(
                 limit=context.number_channel_notices,
                 always_show_viewed=False,
                 silent=True,
             )
+            return_value = func(*args, **kwargs)
+            display_notices(channel_notice_set)
```

Comment:
Let's add a comment here to explain that we first retrieve the notices before calling the decorated function, since it may have side-effects of not doing that.